### PR TITLE
Allow bearer auth against web-config-server

### PR DESCRIPTION
**Pull request by edmofro**
### Issue #:
https://github.com/beyondessential/tupaia-backlog/issues/2583

### Changes:
Adds support for bearer auth

### Manual deploy steps required
- [ ] Add JWT_SECRET to the .env on dev and production

### CODE REVIEW

File: packages/web-config-server/src/authSession/getUserFromAuthHeader.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  if (authHeader.startsWith('Basic ')) {
+    return getUserFromBasicAuth(authenticator, authHeader);
+  }
+
+  if (authHeader.startsWith('Bearer ')) {
+    return getUserFromBearerAuth(models, authHeader);
+  }
+
+  return null;
+};
```
Comment (rohan-bes) [inline packages/web-config-server/src/authSession/getUserFromAuthHeader.js:47]: Would love to migrate web-config-server over to new server-boilerplate whenever there's time 😄 (not suggesting we do this now btw!)

File: packages/lesmis-server/examples.http [hunk 2]
```diff
@@ -41,3 +41,9 @@ Content-Type: {{contentType}}
 GET http://{{host}}/{{version}}/user HTTP/1.1
 content-type: {{contentType}}
 Cookie: {{sessionCookie}}
+
+
+### Logged in reports fetch
+GET http://{{host}}/{{version}}/reportData/LA_Phonhong%20District/Laos_Schools_Number_Of_Schools_By_Type_Table?type=view&dashboardGroupId=337 HTTP/1.1
+content-type: {{contentType}}
+Cookie: {{sessionCookie}}
```
Comment (rohan-bes) [inline packages/lesmis-server/examples.http:49]: Love these! 👍

**Review by rohan-bes — APPROVED** [review]
Looks good!

File: packages/lesmis-server/examples.http (hunk 2 above)
Comment (edmofro) [reply packages/lesmis-server/examples.http:49]: Thanks! Turns out you don't need to explicitly set the Cookie, the api client will take care of that just like a browser when the response from login sets it. Next PR removes them.

File: packages/web-config-server/src/authSession/getUserFromAuthHeader.js (hunk 1 above)
Comment (edmofro) [reply packages/web-config-server/src/authSession/getUserFromAuthHeader.js:47]: Yeah that'd be so nice... 😍

**Comment by edmofro** [comment]
Merged via https://github.com/beyondessential/tupaia/pull/2375

**Pull request closed without merging**